pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -48,12 +48,12 @@ var systemDefaultConstraints = []v1.TopologySpreadConstraint{
 	{
 		TopologyKey:       v1.LabelHostname,
 		WhenUnsatisfiable: v1.ScheduleAnyway,
-		MaxSkew:           3,
+		MaxSkew:           1,
```

Comment:
@alculquicondor what's the original thought of setting the soft maxSkew on host/zone to 3/5 respectively?

I think setting them to 1/1 is more compatible with the original behavior of SelectorSpread. WDYT?

Reply:
For the calculations, maxSkew works kind of like an inverse weight (oversimplifying).

When I run some simulations, this configuration gave a comparatively similar score than the original selectorSpread.

Also, having something different than the minimum possible value, gives chances to users to set a stronger spreading if they need it.

Reply:
> When I run some simulations

I recalled those Python scripts you wrote.

> Also, having something different than the minimum possible value, gives chances to users to set a stronger spreading if they need it.

Setting a default value for score-like functions is always hard... Given maxSkew is a sub-level parameter for plugin PodTopologySpread. I'm more inclined to make it simple (set it to 1), and let the plugin weight be the dominant factor as what other plugins do. So that when user wants to increase the effect of pods spreading, they don't need to change 2 places (plugin's weight and sub-level parameter maxSkew). WDYT?

Also, @Fei-Guo's DP may give us more confidence on setting it to 1.

Reply:
Came across #91258, which is also consistent to make weight as a single dominant factor to impact pods spread behavior.

Reply:
>  So that when user wants to increase the effect of pods spreading, they don't need to change 2 places (plugin's weight and sub-level parameter maxSkew)

These are different APIs for different users. End users don't have access to plugin weight.

Reply:
The context was more based on w/o user's input to specify soft maxSkew on individual workload. So by maxSkew here, I meant the one embedded in plugin's config here:

https://github.com/kubernetes/kubernetes/blob/6673e7a93d8f0e81343309350e3ec7e7afc7fd73/pkg/scheduler/apis/config/v1/defaults.go#L224

which is rendered at:

https://github.com/kubernetes/kubernetes/blob/6673e7a93d8f0e81343309350e3ec7e7afc7fd73/pkg/scheduler/framework/plugins/podtopologyspread/plugin.go#L107-L110

Reply:
That doesn't sound true, the logic is: if a pod carried user-specified soft/hard maxSkew, it always prioritizes over system settings (the if clause below).

https://github.com/kubernetes/kubernetes/blob/6673e7a93d8f0e81343309350e3ec7e7afc7fd73/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L59-L75

So again, the context, or the problem this PR tries to solve is for plain workload w/o any maxSkew specified. For workloads with explicit soft maxSkew, those always win out.

Reply:
>  if a pod carried user-specified soft/hard maxSkew, it always prioritizes over system settings (the if clause below).

That is true. However, my point is that if the admin maxes out the possible value for a system setting, then the end-user has no way of saying: I want **better** spreading.

Reply:
If default schedulable strategy is `DoNotSchedule`, make it 3/5 rather than 1/1 makes more sense to me because maxSkew=1 is too strict, but for soft strategy, I can't say which is better in no uncertain terms, as Wei said, it's always hard to define default value for score functions. In our production, we usually have more than 3 replicas, so we can tolerate maxSkew bigger than 1 because they're already HA to some extend. But for test environment, that's opposite, less replicas there. Sorry to say that, but honestly, the user story mentioned above is not that convincible to me because it's the expected result.

But I think we should update the spec.